python/ql/src/Expressions/HashedButNoHash.ql

```diff
@@ -51,9 +51,31 @@ predicate is_unhashable(ControlFlowNode f, ClassObject cls, ControlFlowNode orig
     )
 }
 
+/**
+ * Holds if `f` is inside a `try` that catches `TypeError`. For example:
+ *
+ *    try:
+ *       ... f ...
+ *    except TypeError:
+ *       ...
+ *
+ * This predicate is used to eliminate false positive results. If `hash`
+ * is called on an unhashable object then a `TypeError` will be thrown.
+ * But this is not a bug if the code catches the `TypeError` and handles
+ * it.
+ */
+predicate typeerror_is_caught(ControlFlowNode f) {
+    exists (Try try, CheckClass c |
```

Comment:
In general, use `ClassObject` not `CheckClass`.
In this case you case refer to the python object `type` directly as `theTypeType()`:
```
try.getAHandler().getType().refersTo(theTypeType())
```

Reply:
Yes, sorry for the confusing comment.